### PR TITLE
[DOP-3866]: Throw warning and ignore nested projects on detection within parser

### PR DESCRIPTION
**Pull request by branberry**
### Ticket 

[DOP-3866](https://jira.mongodb.org/browse/DOP-3866)

### Notes

This PR adds logic to handle nested projects. When parsing a project, we call the `get_file` method and provide the source directory as the root path. If we come across another directory containing a `snooty.toml`, we know that this is a nested project, and we do not traverse that directory.

### CODE REVIEW

File: snooty/test_util.py [hunk 1]
```diff
@@ -75,6 +75,20 @@ def test_get_files() -> None:
     ]
 
 
+def test_get_files_nested() -> None:
+    # The test_data/nested_project paths test ho we handle nested projects within the monorepository.
```
Comment (mmeigs) [inline snooty/test_util.py:79]: how* (I think)

File: snooty/diagnostics.py [hunk 2]
```diff
@@ -883,3 +883,15 @@ def __init__(
...
+class NestedProject(Diagnostic):
+    severity = Diagnostic.Level.warning
+
+    def __init__(
+        self,
+        nested_project: str,
+        start: Union[int, Tuple[int, int]],
+        end: Union[None, int, Tuple[int, int]] = None,
+    ):
+        super().__init__(f"""Nested project detected: {nested_project}""", start, end)
```
Comment (mmeigs) [inline snooty/diagnostics.py:897]: Do we want more info here, as in letting the user know that the parser will not parse the nested project?

File: snooty/util.py [hunk 3]
```diff
@@ -114,13 +119,22 @@ def get_files(
         # Preserve both the actual resolved path and the directory name
         dirs_set = dict(((base_resolved.joinpath(d).resolve(), d) for d in dirs))
 
+        nested_set = set()
+        for d_path, d_name in dirs_set.items():
+            dir_snooty_path = d_path / SNOOTY_TOML
```
Comment (seungpark) [inline snooty/util.py:124]: TIL about truediv

**Review by seungpark — APPROVED** [review]
this LGTM. good tests to showcase pulling files from nested project directories, as well as the diagnostic 👍

File: snooty/util.py (hunk 3 above)
Comment (branberry) [reply snooty/util.py:124]: I actually didn't know about truediv until I googled it just now! Haha. I just saw that we were joining paths elsewhere in the codebase with a forward slash

**Pull request merged**